pkg/cluster/manager/manager.go

```diff
@@ -134,3 +136,54 @@ func (m *Manager) sshTaskBuilder(name string, topo spec.Topology, user string, o
 		).
 		ClusterSSH(topo, user, opts.SSHTimeout, opts.SSHType, topo.BaseTopo().GlobalOptions.SSHType)
 }
+
+func (m *Manager) fillHostArch(s *cliutil.SSHConnectionProps, topo spec.Topology, gOpt *operator.Options, user string) error {
+	hostArch := map[string]string{}
```

Comment:
SSH port should also be part of the unique key, the SSH target might be a NAT device.

Reply:
When reviewing the rest of the code, I found that for the most part, we didn't handle the SSH Port case. 

eg. Shell's execute will set output by the host, if two nodes are under the same host, the later one's output will be overwritten. 

https://github.com/pingcap/tiup/blob/342f0448f9225ce8b87238319e93a659e1e71188/pkg/cluster/task/shell.go#L34-L53